PR: https://github.com/odoo/odoo/pull/

From: ccce07680faa9ebc4712324e0598d63bfcd4008d
From: Gert Pellin
Date: 2019-08-02 06:55:39

Structural Changes: 5
Total Changes: 10

[IMP] point_of_sale: improve views of pos objects

We made here some small changes on the views of different object of POS
module, and set the field journal_ids required in the views, because
launching a POS without payment method won't magically assign one
anymore.

TASK-ID: 2008468

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_order.py
+++ b/addons/point_of_sale/models/pos_order.py
@@ -479,9 +479,9 @@ class PosOrder(models.Model):
         return self.filtered(lambda order: order.state in filter_states)
 
     name = fields.Char(string='Order Ref', required=True, readonly=True, copy=False, default='/')
-    date_order = fields.Datetime(string='Order Date', readonly=True, index=True, default=fields.Datetime.now)
+    date_order = fields.Datetime(string='Date', readonly=True, index=True, default=fields.Datetime.now)
     user_id = fields.Many2one(
-        comodel_name='res.users', string='User',
+        comodel_name='res.users', string='Responsible',
         help="Person who uses the cash register. It can be a reliever, a student or an interim employee.",
         default=lambda self: self.env.uid,
         states={'done': [('readonly', True)], 'invoiced': [('readonly', True)]},
@@ -523,7 +523,7 @@ class PosOrder(models.Model):
     )
     note = fields.Text(string='Internal Notes')
     nb_print = fields.Integer(string='Number of Print', readonly=True, copy=False, default=0)
-    pos_reference = fields.Char(string='Receipt Ref', readonly=True, copy=False)
+    pos_reference = fields.Char(string='PoS Order Reference', readonly=True, copy=False)
     sale_journal = fields.Many2one('account.journal', related='session_id.config_id.journal_id', string='Sales Journal', store=True, readonly=True)
     fiscal_position_id = fields.Many2one(
         comodel_name='account.fiscal.position', string='Fiscal Position',

--- a/addons/point_of_sale/models/pos_session.py
+++ b/addons/point_of_sale/models/pos_session.py
@@ -314,6 +314,10 @@ class PosSession(models.Model):
             'name': _('Orders'),
             'res_model': 'pos.order',
             'view_mode': 'tree,form',
+            'views': [
+                (self.env.ref('point_of_sale.view_pos_order_tree_no_session_id').id, 'tree'),
+                (self.env.ref('point_of_sale.view_pos_pos_form').id, 'form'),
+                ],
             'type': 'ir.actions.act_window',
             'domain': [('session_id', 'in', self.ids)],
         }
